File: Demo-LLP-MIP/Demo-LLP-MIP-PHP/v1/ipay/argument/refund/RefundParams.php
<?php




class RefundParams
{
    public $mch_id;
    public $refund_seqno;
    public $refund_time;
    public $txn_seqno;
    public $txn_date;
    public $refund_reason;
    public $refund_amount;
    public $notify_url;
    public $refund_method_infos;
    public $payee_refund_infos;
    public $divide_refund_infos;

    public function __construct()
    {
        $this->mch_id = null;
        $this->refund_seqno = null;
        $this->refund_time = null;
        $this->txn_seqno = null;
        $this->txn_date = null;
        $this->refund_reason = null;
        $this->refund_amount = null;
        $this->notify_url = null;
        $this->refund_method_infos = null;
        $this->payee_refund_infos = null;
        $this->divide_refund_infos = null;
    }

    public function to_dict() {
        $data = [];
        foreach ($this as $key => $value) {
            if ($value !== null) {
                $data[$key] = $value;
            }
        }
        return json_encode($data);
    }
}

File: Demo-LLP-MIP/Demo-LLP-MIP-Python/v1/ipay/argument/refund/RefundParams.py
from v1.info.RefundMethodInfo import RefundMethodInfo
from v1.info.PayeeRefundInfo import PayeeRefundInfo
from v1.info.DivideRefundInfo import DivideRefundInfo


class RefundParams:
    def __init__(self):
        self.mch_id = None
        self.refund_seqno = None
        self.refund_time = None
        self.txn_seqno = None
        self.txn_date = None
        self.refund_reason = None
        self.refund_amount = None
        self.notify_url = None
        self.refund_method_infos = None  # 假设 RefundMethodInfo 为另一个类
        self.payee_refund_infos = None  # 假设 PayeeRefundInfo 为另一个类
        self.divide_refund_infos = None  # 假设 DivideRefundInfo 为另一个类

    def to_dict(self):
        data = {
            "mch_id": self.mch_id,
            "refund_seqno": self.refund_seqno,
            "refund_time": self.refund_time,
            "txn_seqno": self.txn_seqno,
            "txn_date": self.txn_date,
            "refund_reason": self.refund_reason,
            "refund_amount": self.refund_amount,
            "notify_url": self.notify_url
        }

        if self.refund_method_infos is not None:
            data["refund_method_infos"] = [info.__dict__ for info in self.refund_method_infos]
        if self.payee_refund_infos is not None:
            data["payee_refund_infos"] = [info.__dict__ for info in self.payee_refund_infos]
        if self.divide_refund_infos is not None:
            data["divide_refund_infos"] = [info.__dict__ for info in self.divide_refund_infos]

        return {k: v for k, v in data.items() if v}